wrapper/src/main/java/software/amazon/jdbc/RoundRobinHostSelector.java

```diff
@@ -0,0 +1,196 @@
+      "Comma separated list of database host-weight pairs in the format of `<host>:<weight>`.");
+  public static final AwsWrapperProperty ROUND_ROBIN_DEFAULT_WEIGHT = new AwsWrapperProperty(
+      "roundRobinDefaultWeight", "1",
+      "The default weight for any hosts that have not been configured with the `roundRobinHostWeightPairs` parameter.");
+  public static final String STRATEGY_ROUND_ROBIN = "roundRobin";
+  private static final int DEFAULT_WEIGHT = 1;
+  private static final long DEFAULT_ROUND_ROBIN_CACHE_EXPIRE_NANO = TimeUnit.MINUTES.toNanos(10);
+  static final Pattern HOST_WEIGHT_PAIRS_PATTERN =
+      Pattern.compile(
+          "((?<host>[^:/?#]*):(?<weight>[0-9]*))");
+  private static final CacheMap<String, RoundRobinClusterInfo> roundRobinCache = new CacheMap<>();
+
+  static {
+    PropertyDefinition.registerPluginProperties(RoundRobinHostSelector.class);
+  }
+
+  @Override
+  public synchronized HostSpec getHost(
+      final @NonNull List<HostSpec> hosts,
+      final @NonNull HostRole role,
+      final @NonNull Properties props) throws SQLException {
+    final List<HostSpec> eligibleHosts = hosts.stream()
+        .filter(hostSpec -> role.equals(hostSpec.getRole()))
+        .sorted(Comparator.comparing(HostSpec::getHost))
+        .collect(Collectors.toList());
+
+    if (eligibleHosts.isEmpty()) {
+      throw new SQLException(Messages.get("HostSelector.noHostsMatchingRole", new Object[]{role}));
+    }
+
+    // Create new cache entries for provided hosts if necessary. All hosts point to the same cluster info.
+    createCacheEntryForHosts(eligibleHosts, props);
+    final String currentClusterInfoKey = eligibleHosts.get(0).getHost();
+    final RoundRobinClusterInfo clusterInfo = roundRobinCache.get(currentClusterInfoKey);
+
+    final HostSpec lastHost = clusterInfo.lastHost;
+    int lastHostIndex = -1;
+
+    // Check if lastHost is in list of eligible hosts. Update lastHostIndex.
+    if (lastHost != null) {
+      for (int i = 0; i < eligibleHosts.size(); i++) {
+        if (eligibleHosts.get(i).getHost().equals(lastHost.getHost())) {
+          lastHostIndex = i;
+        }
+      }
+    }
+
+    final int targetHostIndex;
+    // If the host is weighted and the lastHost is in the eligibleHosts list.
+    if (clusterInfo.weightCounter > 0 && lastHostIndex != -1) {
+      targetHostIndex = lastHostIndex;
+    } else {
+      if (lastHostIndex != -1 && lastHostIndex != eligibleHosts.size() - 1) {
+        targetHostIndex = lastHostIndex + 1;
+      } else {
+        targetHostIndex = 0;
+      }
+
+      final Integer weight = clusterInfo.clusterWeightsMap.get(eligibleHosts.get(targetHostIndex).getHost());
+      clusterInfo.weightCounter = weight == null ? clusterInfo.defaultWeight : weight;
+    }
+
+    clusterInfo.weightCounter--;
+    clusterInfo.lastHost = eligibleHosts.get(targetHostIndex);
+
+    return eligibleHosts.get(targetHostIndex);
+  }
+
+  private synchronized void createCacheEntryForHosts(
```

Comment:
This method is private and is used in getHost() that is public and synchronized. You can safely remove "synchronized" here.